core/services/relay/delegate/delegate.go

```diff
@@ -0,0 +1,144 @@
+		err = multierr.Combine(err, r.Close())
+	}
+	return err
+}
+
+// A delegate relayer is healthy if all relayers it manages are ready.
+func (d delegate) Ready() error {
+	var err error
+	for _, r := range d.relayers {
+		err = multierr.Combine(err, r.Ready())
+	}
+	return err
+}
+
+// A delegate relayer is healthy if all relayers it manages are healthy.
+func (d delegate) Healthy() error {
+	var err error
+	for _, r := range d.relayers {
+		err = multierr.Combine(err, r.Healthy())
+	}
+	return err
+}
+
+func (d delegate) NewOCR2Provider(externalJobID uuid.UUID, s interface{}) (relay.OCR2Provider, error) {
+	spec, ok := s.(*job.OffchainReporting2OracleSpec)
+	if !ok {
+		return nil, errors.New("unsuccessful cast to 'job.OffchainReporting2OracleSpec'")
+	}
+
+	choice := spec.Relay
+	switch choice {
+	case relay.Ethereum:
+		var config ethereum.RelayConfig
+		err := json.Unmarshal(spec.RelayConfig.Bytes(), &config)
+		if err != nil {
+			return nil, err
+		}
+
+		return d.relayers[choice].NewOCR2Provider(externalJobID, ethereum.OCR2Spec{
+			ID:             spec.ID,
+			IsBootstrap:    spec.IsBootstrapPeer,
+			ContractID:     spec.ContractID,
+			OCRKeyBundleID: spec.OCRKeyBundleID,
+			TransmitterID:  spec.TransmitterID,
+			ChainID:        config.ChainID.ToInt(),
+		})
+	case relay.Solana:
+		var config solana.RelayConfig
+		err := json.Unmarshal(spec.RelayConfig.Bytes(), &config)
+		if err != nil {
+			return nil, errors.Wrap(err, "error on 'spec.RelayConfig' unmarshal")
+		}
+
+		programID, err := solanaGo.PublicKeyFromBase58(spec.ContractID.ValueOrZero())
+		if err != nil {
+			return nil, errors.Wrap(err, "error on 'solana.PublicKeyFromBase58' for 'spec.ContractID")
+		}
+
+		stateID, err := solanaGo.PublicKeyFromBase58(config.StateID)
+		if err != nil {
+			return nil, errors.Wrap(err, "error on 'solana.PublicKeyFromBase58' for 'spec.RelayConfig.StateID")
+		}
+		validatorProgramID, err := solanaGo.PublicKeyFromBase58(config.ValidatorProgramID)
+		if err != nil {
+			return nil, errors.Wrap(err, "error on 'solana.PublicKeyFromBase58' for 'spec.RelayConfig.StateID")
+		}
+		transmissionsID, err := solanaGo.PublicKeyFromBase58(config.TransmissionsID)
+		if err != nil {
+			return nil, errors.Wrap(err, "error on 'solana.PublicKeyFromBase58' for 'spec.RelayConfig.TransmissionsID")
+		}
+		transmissionSigner, err := d.ks.Solana().Get(transmissionsID.String())
+		if err != nil {
+			return nil, err
+		}
+		return d.relayers[choice].NewOCR2Provider(externalJobID, solana.OCR2Spec{
+			ID:                 spec.ID,
+			IsBootstrap:        spec.IsBootstrapPeer,
+			NodeEndpointRPC:    config.NodeEndpointRPC,
+			NodeEndpointWS:     config.NodeEndpointWS,
+			ProgramID:          programID,
+			StateID:            stateID,
+			ValidatorProgramID: validatorProgramID,
+			TransmissionsID:    transmissionsID,
+			TransmissionSigner: transmissionSigner,
+		})
+	default:
+		return nil, fmt.Errorf("unknown relayer network type: %s", spec.Relay)
```

Comment:
Please always use `errors.Errorf` because it contains additional data like stack traces